### PR TITLE
Make Slider change the value with onKeyEvent for: all arrows, PgDn, PdUp, Home, End buttons

### PR DESCRIPTION
**Pull request by eymar**
WIP: if the change is fine from the implementation point of view, I'll add some tests

### CODE REVIEW

File: compose/material/material/src/commonMain/kotlin/androidx/compose/material/Slider.kt [hunk 1]
```diff
@@ -245,6 +254,64 @@ fun Slider(
...
+): Modifier {
+    require(steps >= 0) { "steps should be >= 0" }
+
+    return this.onKeyEvent {
+        if (it.type != KeyEventType.KeyDown || !enabled) return@onKeyEvent false
+        val rangeLength = abs(valueRange.endInclusive - valueRange.start)
+        // When steps == 0, it means that a user is not limited by a step length (delta) when using touch or mouse.
+        // But it is not possible to adjust the value continuously when using keyboard buttons -
+        // the delta has to be discrete. In this case, 1% of the valueRange seems to make sense.
+        val delta = rangeLength / if (steps > 0) steps + 2 else 100
```
Comment (igordmn) [inline compose/material/material/src/commonMain/kotlin/androidx/compose/material/Slider.kt]: It looks like it should be `steps + 1`. I tested it with Slider in run1, when set `steps = 3`

**Comment by igordmn** [comment]
Thanks! Looks good, except the small issue

File: compose/material/material/src/commonMain/kotlin/androidx/compose/material/Slider.kt [hunk 2]
```diff
@@ -245,6 +254,64 @@ fun Slider(
...
 }
 
+@OptIn(ExperimentalComposeUiApi::class)
+private fun Modifier.slideOnKeyEvents(
+    enabled: Boolean,
+    steps: Int,
+    valueRange: ClosedFloatingPointRange<Float>,
+    value: Float,
+    isRtl: Boolean,
+    onValueChangeState: State<(Float) -> Unit>
```
Comment (dima-avdeev-jb) [inline compose/material/material/src/commonMain/kotlin/androidx/compose/material/Slider.kt:264]: It is possible to contains closure inside State<...> ?
Does closure have proper equals() implementation?

Comment (eymar) [reply compose/material/material/src/commonMain/kotlin/androidx/compose/material/Slider.kt:264]: yes, it possible to store it in State. They're equal when the reference is the same.

But in our case, we don't care if they are equal or not. We use `fun rememberUpdateState(): State<T>` which creates and stores an instance of MutableState and then updates the underlying value on every recomposition.

**Review by dima-avdeev-jb — APPROVED** [review]

**Comment by eymar** [comment]
Added the tests

**Review by igordmn — APPROVED** [review]

**Pull request merged**